crates/cargo-contract/src/cmd/config.rs

```diff
@@ -143,73 +138,75 @@ impl SignerConfig<Self> for Polkadot {
 
 /// Struct representing the implementation of the sr25519 signer
 #[derive(Clone)]
-pub struct SignerSR25519<C: Config>(pub PairSigner<C, sp_core::sr25519::Pair>);
+pub struct SignerSR25519<C: Config> {
+    account_id: <C as Config>::AccountId,
+    signer: sp_core::sr25519::Pair,
+}
 
-impl<C: Config> FromStr for SignerSR25519<C>
-where
-    <C as Config>::AccountId: From<sp_core::crypto::AccountId32>,
-{
+impl<C: Config<AccountId: From<AccountId32>>> SignerSR25519<C> {
+    /// Creates a new [`SignerSR25519`] from a [`sp_core::sr25519::Pair`].
+    pub fn new(signer: sp_core::sr25519::Pair) -> Self {
+        Self {
+            account_id: AccountId32(signer.public().0).into(),
```

Comment:
Uses the same implementation from https://github.com/paritytech/polkadot-sdk/blob/c4ba6e0288a127f7f452e09e872960837aa313c0/substrate/primitives/runtime/src/lib.rs#L399, but without taking a dependency for a single conversion function.